aspnetcore/mvc/controllers/actions.md

```diff
@@ -14,80 +14,94 @@ uid: mvc/controllers/actions
 
-* Are located in the root-level "Controllers" folder
+Additional configuration is required if these two conventions are disregarded.
 
-* Inherit from Microsoft.AspNetCore.Mvc.Controller
+Controllers should follow the [Explicit Dependencies Principle](http://deviq.com/explicit-dependencies-principle) and request any dependencies their actions require through their constructor using [constructor injection](xref:mvc/controllers/dependency-injection#constructor-injection).
 
-These two conventions are not required.
-
-Within the Model-View-Controller pattern, a Controller is responsible for the initial processing of the request and instantiation of the Model. Generally, business decisions should  be performed within the Model.
+Within the **M**odel-**V**iew-**C**ontroller pattern, a controller is responsible for the initial processing of the request and instantiation of the model. Generally, business decisions should be performed within the model.
 
 > [!NOTE]
-> The Model should be a *Plain Old CLR Object (POCO)*, not a `DbContext` or database-related type.
+> The model should be a *Plain Old CLR Object (POCO)*, not a `DbContext` object or a database-related type.
 
-The controller takes the result of the model's processing (if any), returns the proper view along with the associated view data. Learn more: [Overview of ASP.NET Core MVC](../overview.md) and [Getting started with ASP.NET Core MVC and Visual Studio](../../tutorials/first-mvc-app/start-mvc.md).
+The controller takes the result of the model's processing (if any) and returns the proper view along with the associated view data. Learn more at [Overview of ASP.NET Core MVC](xref:mvc/overview) and [Getting started with ASP.NET Core MVC and Visual Studio](xref:tutorials/first-mvc-app/start-mvc).
 
->[!TIP]
-> The Controller is a *UI level* abstraction. Its responsibility is to ensure incoming request data is valid and to choose which view (or result for an API) should be returned. In well-factored apps it will not directly include data access or business logic, but instead will delegate to services handling these responsibilities.
+> [!TIP]
+> The controller is a *UI-level* abstraction. Its responsibilities are to ensure incoming request data is valid and to choose which view (or result for an API) should be returned. In well-factored apps, it does not directly include data access or business logic. Instead, the controller delegates to services handling these responsibilities.
 
 ## Defining Actions
 
-Any public method on a controller type is an action. Parameters on actions are bound to request data and validated using [model binding](../models/model-binding.md).
+Any public method on a controller type is an action. Parameters on actions are bound to request data and are validated using [model binding](xref:mvc/models/model-binding).
 
->[!WARNING]
+> [!WARNING]
 > Action methods that accept parameters should verify the `ModelState.IsValid` property is true.
```

Comment:
This seems out of place here and I don't think in any case that it really needs to be an obtrusive "warning" banner.  Also "that accept parameters" may be misleading, as model binding data can come from form data that isn't passed in as a method parameter.